compiler/rustc_infer/src/infer/type_variable.rs

```diff
@@ -238,6 +238,9 @@ impl<'tcx> ut::UnifyKey for TyVidEqKey<'tcx> {
     fn tag() -> &'static str {
         "TyVidEqKey"
     }
+    fn order_roots(a: Self, _: &Self::Value, b: Self, _: &Self::Value) -> Option<(Self, Self)> {
```

Comment:
how hot are these functions, could we get even more perf by adding inline to them? I would certainly assume that we already inline everywhere we can

Reply:
I hope they're already inlined b/c they're small and/or b/c PGO inlines them. I think #142088 will make this optimization more worthwhile, so I'll rerun this when that one lands 🤔